extractjava.py

```diff
@@ -0,0 +1,43 @@
+import sys, re, os, json
+from grammars.grammars import java
+from extractor import find_subtree, remove_tabs_newlines
+
+import re
+RE_CMT = re.compile(r"}(\s|\\r)*/\*.*?\*/$", re.MULTILINE)
```

Comment:
It feels odd that this regex starts with `}`. Also, I was sort-of expecting it to deal with `//` comments too?

Reply:
The reason for this is, that we always attach whitespace/comments to the previous token. So if we have
```
public void foo(){}
/** Some doc comment **/
public void bar(){}
```
The the comment is attached to the `}` of the `foo` function. However, I don't want to remove all comments inside of functions blocks, so I only remove block comments coming after `}` and replace the whole match with `}` (I guess this will still remove comments that come after a if-blocks etc., but we can live with that). I don't have encountered `//` comments after a function block, so there was no need to get rid of those.

Reply:
Dumb question: why are we bothered with comments at all?

Reply:
In the javastdlib, these doc comments are often quite large, which meant the the experiment would slow down to a crawl, as each character is typed one at a time (also: an open `/*` would relex the entire program every time as well, slowing things down even more). We can probably put them back in, in the final run, though I don't think it makes a difference.

Reply:
Ah, are we using this to remove comments?

Reply:
Yes, in the example above, this would remove the comment from the `foo` functions, before embedding the function into another language.